velox/exec/tests/ExpressionBuilderTest.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/tests/utils/ExpressionBuilder.h"
+
+#include <gtest/gtest.h>
+#include "velox/parse/ExpressionsParser.h"
+#include "velox/type/Variant.h"
+
+namespace facebook::velox::expr::builder::test {
+namespace {
+
+// Test convenience functions for downcasting.
+template <typename T>
+std::shared_ptr<const T> as(ExprWrapper in) {
+  return std::dynamic_pointer_cast<const T>(in.ptr);
+}
+
+template <typename T>
+bool is(ExprWrapper in) {
+  return std::dynamic_pointer_cast<const T>(in.ptr) != nullptr;
+}
+
+// Parses a SQL expression using DuckDB.
+auto parseSql(const std::string& sql) {
+  return parse::parseExpr(sql, {});
+}
+
+TEST(ExpressionBuilderTest, fieldAccess) {
+  ExprWrapper result;
+
+  result = field("col");
```

Comment:
I noticed that applications using Velox library often do not have the luxury of using Velox symbols unqualified and forced to use velox::BIGINT() vs. just BIGINT(). 

Since this API is all about brevity and readability, how do you think applications can avoid having to prefix all these names with velox::expr::builder?

```
velox::expr::builder::field("f") vs. field("f") 
```

Reply:
Good question. We can naturally move this to the top-level namespace. My concern was to prevent inadvertent symbols clashes by requiring API users to explicitly import namespace. For example, maybe there could be other `velox::field()` symbols defined somewhere else? This was out of abundant caution though; more likely than not moving to top-level should be fine.

Reply:
Curious what is the reason preventing applications from doing `using namespace velox::expr::builder`.  `using namespace velox` is probably a bit too broad but `using namespace velox::expr::builder` probably should be fine.